packages/flutter_tools/lib/src/android/device_android.dart

```diff
@@ -529,3 +511,38 @@ String getAdbPath() {
     return _defaultAdbPath;
   }
 }
+
+class _AdbLogReader extends DeviceLogReader {
+  _AdbLogReader(this.device);
+
+  final AndroidDevice device;
+
+  String get name => 'Android';
+
+  Future<int> logs({bool clear: false}) async {
+    if (!device.isConnected())
+      return 2;
+
+    if (clear)
+      device.clearLogs();
+
+    return await runCommandAndStreamOutput(device.adbCommandForDevice([
```

Comment:
done, but we may want to make a pass through this source base and add these in one PR.